File: sqry-lang-python/tests/fixtures/localvars/advanced.py
# Advanced: closures, destructuring, comprehensions, except, with

def closure_capture():
    x = 10
    f = lambda y: x + y
    return f(1)

def destructuring_tuple():
    pair = (1, 2)
    a, b = pair
    return a + b

def destructuring_list():
    data = [1, 2, 3]
    first, *rest = data
    return first

def try_except_binding():
    try:
        result = 1 / 0
    except ZeroDivisionError as err:
        msg = str(err)
    return msg

def with_binding():
    with open("test.txt") as f:
        data = f.read()
    return data

def walrus_operator():
    values = [1, 2, 3, 4, 5]
    filtered = [y for x in values if (y := x * 2) > 4]
    return filtered

def nested_function():
    outer = 10
    def inner():
        return outer + 1
    return inner()

def comprehension_var():
    items = [1, 2, 3]
    result = [item * 2 for item in items]
    return result

File: sqry-lang-rust/tests/fixtures/localvars/advanced.rs
// Advanced: closures, match, if-let, destructuring

fn closure_capture() {
    let x = 10;
    let f = |y| x + y;
    println!("{}", f(1));
}

fn match_binding() {
    let value = Some(42);
    match value {
        Some(inner) => {
            let _result = inner + 1;
        }
        None => {}
    }
}

fn if_let_binding() {
    let value = Some(42);
    if let Some(inner) = value {
        let _result = inner + 1;
    }
}

fn destructuring_tuple() {
    let pair = (1, 2);
    let (a, b) = pair;
    println!("{} {}", a, b);
}

fn destructuring_struct() {
    struct Point { x: i32, y: i32 }
    let p = Point { x: 10, y: 20 };
    let Point { x, y } = p;
    println!("{} {}", x, y);
}

fn shadowing_idiomatic() {
    let x = "hello";
    let x = x.len();
    println!("{}", x);
}

fn unsafe_block_var() {
    unsafe {
        let raw = 42;
        println!("{}", raw);
    }
}
